File: server/Models/Cryptid.cs
using System.ComponentModel.DataAnnotations;

namespace cryptid_book.Models;

public class Cryptid : RepoItem<int>
{
  public string Name { get; set; }
  [Range(0, 10)]
  public int ThreatLevel { get; set; }
  [MaxLength(3000)]
  public string ImgUrl { get; set; }
  public string Origin { get; set; }
  public string Size { get; set; }
  public string CryptidCode { get; set; } = Guid.NewGuid().ToString();
  public string DiscovererId { get; set; }
  public Profile Discoverer { get; set; }
}

File: client/src/models/Cryptid.js
import { Profile } from "./Profile.js"

export class Cryptid {

  constructor(data) {
    this.id = data.id
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
    this.name = data.name
    this.threatLevel = data.threatLevel
    this.imgUrl = data.imgUrl
    this.origin = data.origin
    this.size = data.size
    this.cryptidCode = data.cryptidCode
    this.discovererId = data.discovererId
    this.discoverer = new Profile(data.discoverer)
  }
}